multicast/recv.py

```diff
@@ -104,34 +89,172 @@
 		from . import __MCAST_DEFAULT_PORT as __MCAST_DEFAULT_PORT
 	else:  # pragma: no branch
 		__MCAST_DEFAULT_PORT = sys.modules["""multicast.__MCAST_DEFAULT_PORT"""]
-except Exception as importErr:
+except Exception as importErr:  # pragma: no branch
 	del importErr
 	import multicast.__MCAST_DEFAULT_PORT as __MCAST_DEFAULT_PORT
 
 
+def genSocket():
+	"""Will generate an unbound socket.socket object ready to receive network traffic.
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+		>>> import multicast
+		>>> multicast.__doc__ is not None
+		True
+		>>>
+
+	Testcase 0: Recv should be automaticly imported.
+		A: Test that the multicast component is initialized.
+		B: Test that the recv component is initialized.
+
+		>>> multicast is not None
+		True
+		>>> multicast.recv is not None
+		True
+		>>>
+
+	Testcase 1: Recv should have genSocket() function that returns a socket.socket object.
+		A: Test that the recv component has the function 'genSocket'
+		B: Test that the 'genSocket' function returns a socket
+
+		>>> multicast.recv.genSocket is not None
+		True
+		>>> multicast.recv.genSocket #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<function genSocket at ...>
+		>>> type(multicast.recv.genSocket)
+		<class 'function'>
+		>>> type(multicast.recv.genSocket())
+		<class 'socket.socket'>
+		>>>
+
+
+	"""
+	sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+	# allow reuse of socket (to allow another instance of python running this
+	# script binding to the same ip/port)
```

Comment:
🤔 might be better as documentation then just comments
```suggestion
	"""Allow reuse of socket (to allow another instance of python running this
	script binding to the same ip/port)"""
```

Reply:
:info: can be covered as part of #33